x-pack/plugins/apm/server/lib/search_strategies/correlations/query_correlation.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+
+import type { ElasticsearchClient } from 'src/core/server';
+
+import { TRANSACTION_DURATION } from '../../../../common/elasticsearch_fieldnames';
+import type { SearchServiceParams } from '../../../../common/search_strategies/correlations/types';
+
+import { getQueryWithParams } from './get_query_with_params';
+import { PERCENTILES_STEP } from './constants';
+
+export interface HistogramItem {
+  key: number;
+  doc_count: number;
+}
+
+interface ResponseHitSource {
+  [s: string]: unknown;
+}
+interface ResponseHit {
+  _source: ResponseHitSource;
+}
+
+interface BucketCorrelation {
+  buckets_path: string;
+  function: {
+    count_correlation: {
+      indicator: {
+        expectations: number[];
+        doc_count: number;
+      };
+    };
+  };
+}
+
+export const getTransactionDurationCorrelationRequest = (
+  params: SearchServiceParams,
+  percentiles: number[],
+  totalHits: number,
+  fieldName?: string,
+  fieldValue?: string
+): estypes.SearchRequest => {
+  const query = getQueryWithParams(params);
+
+  if (typeof fieldName === 'string' && typeof fieldValue === 'string') {
```

Comment:
No need for `typeof`

Reply:
Updated here https://github.com/elastic/kibana/pull/99905/commits/7dee7b41cc2dd243dcde7e5a5660c9e5b40532c3